pg_search/src/postgres/utils.rs

```diff
@@ -156,30 +155,18 @@ impl VisibilityChecker {
 
             pg_sys::LockBuffer(self.last_buffer, pg_sys::BUFFER_LOCK_SHARE as i32);
 
-            self.check_page_vis(self.ipd, offsetno, self.last_buffer)
+            self.check_page_vis(self.last_buffer)
         }
     }
 
-    unsafe fn check_page_vis(
-        &mut self,
-        mut item_pointer: ItemPointerData,
-        offsetno: OffsetNumber,
-        buffer: Buffer,
-    ) -> bool {
+    unsafe fn check_page_vis(&mut self, buffer: Buffer) -> bool {
         unsafe {
-            let page = pg_sys::BufferGetPage(buffer);
-            let item_id = pg_sys::PageGetItemId(page, offsetno);
-            let mut heap_tuple = pg_sys::HeapTupleData {
-                t_data: pg_sys::PageGetItem(page, item_id) as pg_sys::HeapTupleHeader,
-                t_len: item_id.as_ref().unwrap().lp_len(),
-                t_tableOid: (*self.relation).rd_id,
-                t_self: item_pointer,
-            };
+            let mut heap_tuple = pg_sys::HeapTupleData::default();
```

Comment:
Does heap tuple not have to be populated?

Reply:
nope.  I just updated the description here, but the `pg_sys: heap_hot_search_buffer()` function does the populating!